send_stream.go

```diff
@@ -673,6 +686,8 @@ func (s *sendStreamAckHandler) OnLost(f wire.Frame) {
 	// If the reliable size was 0 when the stream was cancelled,
 	// the number of outstanding frames was immediately set to 0, and the retransmission queue was dropped.
 	if s.resetErr != nil && (*SendStream)(s).reliableOffset() == 0 {
+		// Return the frame to pool since it won't be retransmitted
+		sf.PutBack()
```

Comment:
This should be really rare, and is one of the cases where I'd opt for simplicity.

Reply:
You don't need to remove it though, this is pretty straight-forward.